gc/base/MemorySubSpace.cpp

```diff
@@ -936,14 +936,22 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 		reportSystemGCEnd(env);
 		env->releaseExclusiveVMAccessForGC();
 
+		bool isCheckpointGC = (J9MMCONSTANT_EXPLICIT_GC_PREPARE_FOR_CHECKPOINT == gcCode);
 
-		if ((J9MMCONSTANT_EXPLICIT_GC_PREPARE_FOR_CHECKPOINT == gcCode)
+		if (isCheckpointGC
 #if defined(OMR_GC_IDLE_HEAP_MANAGER)
 		|| ((J9MMCONSTANT_EXPLICIT_GC_IDLE_GC == gcCode) && (_extensions->gcOnIdle))
 #endif
 		) {
 			OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
 			uint64_t startTime = omrtime_hires_clock();
+
+			uintptr_t includeMemoryType = MEMORY_TYPE_OLD;
+
+			if (isCheckpointGC) {
+				includeMemoryType |= MEMORY_TYPE_NEW;
+			}
+
 			uintptr_t releasedBytes = _extensions->heap->getDefaultMemorySpace()->releaseFreeMemoryPages(env);
```

Comment:
you'll have to modify releaseFreeMemoryPages for  Heap and MemorySpace ,too. 
they  just need to pass the request down to its children